src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenStackAllocInitializerTests.cs

```diff
@@ -2,13 +2,91 @@
 
 using Microsoft.CodeAnalysis.CSharp.Test.Utilities;
 using Microsoft.CodeAnalysis.Test.Utilities;
+using Roslyn.Test.Utilities;
 using Xunit;
 
 namespace Microsoft.CodeAnalysis.CSharp.UnitTests
 {
     [CompilerTrait(CompilerFeature.StackAllocInitializer)]
     public class CodeGenStackAllocInitializerTests : CompilingTestBase
     {
+        [Fact]
+        [WorkItem(29092, "https://github.com/dotnet/roslyn/issues/29092")]
+        public void TestMixedWithInitBlock()
+        {
+
+            var text = @"
+using System;
+
+class Program
+{
+    static void Main(string[] args)
+    {
+        MakeBlock(1, 2, 3);
+    }
+
+    static unsafe void MakeBlock(int a, int b, int c)
+    {
+        int* ptr = stackalloc int[]
+        {
+           0, 0, 0, a, b, c
+        };
+        PrintBytes(ptr, 6);
+    }
+
+    static unsafe void PrintBytes(int* ptr, int count)
+    {
+        for (int i = 0; i < count; i++)
+        {
+            Console.Write(ptr[i]);
+        }
+    }
+}";
+            CompileAndVerify(text,
+                parseOptions: TestOptions.Regular.WithLanguageVersion(LanguageVersion.CSharp7_3),
+                options: TestOptions.UnsafeReleaseExe,
+                expectedOutput: "000123",
+                verify: Verification.Fails).VerifyIL("Program.MakeBlock",
+@"{
+  // Code size       42 (0x2a)
+  .maxstack  4
+  IL_0000:  ldc.i4.s   24
+  IL_0002:  conv.u
+  IL_0003:  localloc
+  IL_0005:  dup
+  IL_0006:  ldc.i4.0
+  IL_0007:  ldc.i4.s   24
+  IL_0009:  initblk
```

Comment:
initblk may not be needed here at all. I think we can revise once https://github.com/dotnet/roslyn/issues/25780 and https://github.com/dotnet/corefx/issues/26948 goes in.

Reply:
As I pointed out in the original issue, as long as locals init is set, this is redundant. I think `initblk` should only be emitted, if `SkipLocalsInit` is applied.

Reply:
Note: we emit initblk if all the bytes are the same e.g. `stackalloc[] {-1,-1,-1}` not solely to "initilaize" the localloc. still, the case with all zeroes is a special case which could be avoided in the absence of SkipLocalsInit.

Reply:
Note that you may go through initialization more than once. SkipLocalsInit will be irrelevant in that case. `initblk` would still be necessary.

Example:
```C#
    static unsafe void MakeBlock(int a, int b, int c)
    {
label1:

        int* ptr = stackalloc int[]
        {
           0, 0, 0, a, b, c
        };

        PrintBytes(ptr, 6);

ptr[0] = 123;

        PrintBytes(ptr, 6);

goto label1;

    }
```

I would not be worried about eliding `initblk` here. 
Just let the user to apply `SkipLocalsInit` if he cares.